doc/en/source/ngsildbasediotintegration.rst

```diff
@@ -0,0 +1,476 @@
+
+
+.. figure:: figures/water_op.png
+
+
+2. Register the following docker image, corresponding to the operator created in above step.
+   
+   
+.. code-block:: console
+
+	fogflow/waterpump
+
+        The above operator send a "off" command after 1 hour if waterPump is "on".
+
+   	To register the image, select DockerImage from left menu, under Operator Registry from dashboard and click register button.
+
+
+.. figure:: figures/waterdocker.png
+
+
+3. Now, to create a fogfunction, toggle on Fog Function in the horizontal bar on dashboard. select Fog Function from left menu and click register button. Enter the name and description (optional) as shown. Right click in the workspace and select task and Entity stream from drop down list and configure details.
+
+
+.. figure:: figures/waterfog.png
+
+
+Note: For a details on fogfunction creation follow the `Document link`_ .
+ 
+.. _`Document link`: https://fogflow.readthedocs.io/en/latest/intent_based_program.html
+
+
+How to Push the Generated Result back to the IoT Agent
+=============================================================
+
+**Step 9**: Fog Function do some data analytics in step no. 4 and publish the analytics result on fogflow broker. Orion-LD  subscribes fogFlow broker for getting the analytics result and orion broker notify the result to the IoT agent.
+
+.. code-block:: console
+
+        curl -iX POST \
+                  'http://localhost:8070/ngsi-ld/v1/subscriptions/' \
+                  -H 'Content-Type: application/json' \
+		  -H 'Integration: IoTI' \
+                  -H 'Accept: application/ld+json' \
+		  -H 'fiware-service: openiot' \
+		  -H 'fiware-servicepath: /' \
+                  -H 'Link: <https://fiware.github.io/data-models/context.jsonld>; rel="https://uri.etsi.org/ngsi-ld/v1/ngsi-ld-core-context.jsonld"; type="application/ld+json"' \
+                  -d ' {
+                        "type": "Subscription",
+                        "entities": [{
+			       "id": "urn:ngsi-ld:Device:water001",
+                               "type": "daresult"
+                        }],
+                      "notification": {
+                          "format": "normalized",
+                          "endpoint": {
+                                   "uri": "http://<orion-ld-brokerIP>:1026",
+                                   "accept": "application/ld+json"
+                           }
+                       }
+                   }'
+
+Note: Replace localhost with IP where Fogflow thinbroker is running and <orion-ld-brokerIP> with IP where orion-ld broker is running.
+ 
+**Step 10**:Thinbroker will notify the analytical data to NGSILD broker as in step No 5, NGSI-LD broker has subscribed for the analytical data.
+
+
+**Step 11**:Use the CURL command to query entities of type "Device" from NGSILD broker.
```

Comment:
instead of NGSI-LD broker it should be Orion broker.

Reply:
Done